mm2src/lp_swap/maker_swap.rs

```diff
@@ -267,16 +306,49 @@ impl MakerSwap {
             taker_payment_confirmed: AtomicBool::new(false),
             conf_settings,
             payment_locktime,
+            p2p_privkey,
             mutable: RwLock::new(MakerSwapMut {
                 data: MakerSwapData::default(),
-                other_persistent_pub: H264::default(),
+                other_maker_coin_htlc_pub: H264::default(),
+                other_taker_coin_htlc_pub: H264::default(),
                 taker_fee: None,
                 maker_payment: None,
                 taker_payment: None,
                 taker_payment_spend: None,
                 maker_payment_refund: None,
                 taker_payment_spend_confirmed: false,
+                my_maker_coin_htlc_keypair: ctx.secp256k1_key_pair().clone(),
+                my_taker_coin_htlc_keypair: ctx.secp256k1_key_pair().clone(),
             }),
+            ctx,
+        }
+    }
+
+    fn get_my_negotiation_data(&self) -> NegotiationDataMsg {
+        let r = self.r();
+        let secret_hash = dhash160(&r.data.secret.0).take().to_vec();
+        let maker_coin_swap_contract = self.maker_coin.swap_contract_address().map_or(vec![], |addr| addr.0);
+        let taker_coin_swap_contract = self.taker_coin.swap_contract_address().map_or(vec![], |addr| addr.0);
+
+        if r.my_maker_coin_htlc_keypair != r.my_taker_coin_htlc_keypair {
```

Comment:
I think we should use the latest version of the negotiation data.
What if we want to add or change a field in the negotiation data. Currently, I see only one way: Add `NegotiationDataV2.1` and `NegotiationDataV3.1` that will be used accordingly to this condition:
`if r.my_maker_coin_htlc_keypair != r.my_taker_coin_htlc_keypair`
I'd suggest just using `NegotiationDataV3` only. On the other hand, it will break the backward compatibility with the old nodes that don't support V3 yet. Or return `HtlcPubkeyData` 😅
I think renaming `NegotiationDataV3` to `NegotiationDataPubkeysPerCoin` is also ok.

Reply:
`NegotiationDataPubkeysPerCoin` will break a naming convention that we already have (`V1`, `V2`). If we need to add new fields, we will add `NegotiationDataV4`. It won't be mandatorily tied to `if r.my_maker_coin_htlc_keypair != r.my_taker_coin_htlc_keypair`, we will maybe have another condition or make it compatible with `NegotiationDataV3 `. Also, when the majority of nodes are updated, we can remove the condition and make the `V3` default for everyone.

According to this, I don't think we need to change the existing code *now* (but we will change it later).